src/services/users.js

```diff
@@ -0,0 +1,178 @@
+
+    const query = {};
+    if (_id) query._id = _id;
+    if (name) query.name = name;
+    if (email) query.email = email;
+    if (status) query.status = status;
+    if (role) query.role = role;
+    if (skip) query.skip = skip;
+    if (take) query.take = take;
+
+    try {
+      const users = await User.find(query)
+        .select('-password')
+        .skip(+query.skip || 0)
+        .limit(+query.take || 50)
+        .exec();
+
+      const count = await User.find(query).countDocuments().exec();
+
+      return {
+        total_size: count,
+        list: users ? res.json(users) : res.json([])
+      };
+    } catch (err) {
+      return res.boom.badRequest();
+    }
+  }
+
+  async updateUser (req, res) {
+    try {
+      const user = await User.findOne({ _id: req.params.id }).exec();
+      const currentStatus = user.status;
+      const statusActive = await UserStatuses.findOne({
+        name: 'Active'
+      }).exec();
+      if (currentStatus.toString() === statusActive._id.toString()) {
+        const updated = await User.findByIdAndUpdate(req.params.id, req.body, {
+          new: true
+        }).exec();
+        return updated ? res.json(true) : res.boom.notFound();
+      } else {
+        res.boom.badRequest('user is not active');
+      }
+    } catch (err) {
+      return res.boom.notFound();
+    }
+  }
+
+  async updateUserPassword (req, res) {
+    try {
+      const user = await User.findOne({ _id: req.params.id }).exec();
+      const currentStatus = user.status;
+      const statusActive = await UserStatuses.findOne({
+        name: 'Active'
+      }).exec();
+      if (currentStatus.toString() === statusActive._id.toString()) {
+        const { oldPassword, newPassword } = req.body;
+        const user = await User.findOne({ _id: req.params.id }).exec();
+        const isMatchPasswords = await bcrypt.compare(
+          oldPassword,
+          user.password
+        );
+        if (!isMatchPasswords) {
+          return res.boom.badRequest('wrong old password');
+        }
+        if (isMatchPasswords && oldPassword !== newPassword) {
+          user.password = newPassword;
+          const salt = await bcrypt.genSalt(10);
+          user.password = await bcrypt.hash(user.password, salt);
+          const updatedUserPassword = User.findOneAndUpdate(
+            { _id: req.params.id },
+            user
+          );
+          return updatedUserPassword ? res.json(true) : res.boom.notFound();
+        } else {
+          res.boom.badRequest('new password cannot be the same as old');
+        }
+      }
+    } catch {
+      res.boom.notFound();
+    }
+  }
+
+  async blockUser (req, res) {
+    try {
+      const statusBlocked = await UserStatuses.findOne({
+        name: 'Blocked'
+      }).exec();
+      const statusDeleted = await UserStatuses.findOne({
+        name: 'Deleted'
+      }).exec();
+      const user = await User.findOne({ _id: req.params.id }).exec();
+      const currentStatus = user.status;
```

Comment:
Throw not found error here if user doesn't exist

Reply:
fixed